extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -460,20 +460,25 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     if (this.adPageIds_[pageId]) {
       // We are switching to an ad.
       const adIndex = this.adPageIds_[pageId];
+      const adPage = this.adPages_[adIndex - 1];
+
+      if (!adPage.hasBeenViewed()) {
+        this.pendingAdView_ = false;
+        this.resetPageCount_();
```

Comment:
Might not matter after all, but maybe we should resetPageCount within startNextAdPage?

Reply:
Thats where it was before :), I moved it because sometimes I need to reset the counter without starting the next ad loading

Reply:
If there are not enough pages left, it would still reset the page count. I felt it might not matter. Does it?

Reply:
If there are not enough pages left, it would still reset the page count. I felt it might not matter. Does it?

Reply:
If there are not enough pages left, it would still reset the page count. I felt it might not matter. Does it?

Reply:
Oh I see, I think I need to reset because if the count > the interval to place ads it will keep thinking it needs to place an ad. I maybe should have a signal that just says STOP. I want to do a page manager class that simplifies this logic. Coming soon... :)